manager/manager.go

```diff
@@ -346,19 +363,34 @@ func (m *Manager) Run(parent context.Context) error {
 	// Set the raft server as serving for the health server
 	healthServer.SetServingStatus("Raft", api.HealthCheckResponse_SERVING)
 
-	if err := m.raftNode.JoinAndStart(ctx); err != nil {
+	if err := m.joinAndStart(ctx); err != nil {
 		return errors.Wrap(err, "can't initialize raft node")
 	}
+	m.keyRotator.SetKeyWriter(m.config.SecurityConfig.KeyWriter())
 
 	localHealthServer.SetServingStatus("ControlAPI", api.HealthCheckResponse_SERVING)
 
 	close(m.started)
 
+	go m.watchForKEKChanges(ctx)
+
 	go func() {
+		if m.keyRotator.GetNewKey() != nil {
+			m.keyRotator.rotationChan <- struct{}{}
+		}
 		err := m.raftNode.Run(ctx)
 		if err != nil {
 			log.G(ctx).Error(err)
 			m.Stop(ctx)
+			if err == raft.ErrMemberRemoved {
+				if err := m.config.SecurityConfig.KeyWriter().UpdateHeaders(func(headers map[string]string, _ []byte) error {
+					delete(headers, defaultRaftDEKKey)
+					delete(headers, defaultRaftDekKeyPending)
+					return nil
+				}); err != nil {
+					log.G(ctx).Error(errors.Wrap(err, "unable to remove raft DEK from TLS certificate"))
+				}
+			}
```

Comment:
Not sure if all of this is necessary and worth the complexity.

If our role changed to worker, we'll renew the certificate and the new cert/key won't have the DEK stored inside it, right?

I think it's cleaner for the presence of the DEK to be tied to the role (TLS cert OU) and ignore other factors such as this error from raft.